pkg/migrate/cstor/cstorvolumepolicy.go

```diff
@@ -0,0 +1,162 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type config struct {
+	Name  string `yaml:"name"`
+	Value string `yaml:"value"`
+}
+
+type quantity struct {
+	Memory string `yaml:"memory,omitempty"`
+	CPU    string `yaml:"cpu,omitempty"`
+}
+
+func (v *VolumeMigrator) createCVPforConfig(sc *storagev1.StorageClass) error {
+	_, err := v.OpenebsClientset.CstorV1().
+		CStorVolumePolicies(v.OpenebsNamespace).
+		Get(sc.Name, metav1.GetOptions{})
+	found := false
+	if err == nil {
+		found = true
+	}
+	if !k8serrors.IsNotFound(err) {
+		return err
+	}
+	scConfig := sc.Annotations["cas.openebs.io/config"]
+	scConfig = strings.TrimSpace(scConfig)
+	configs := []config{}
+	err = yaml.Unmarshal([]byte(scConfig), &configs)
+	if err != nil {
+		return err
+	}
+	cvp := &cstor.CStorVolumePolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sc.Name,
+			Namespace: v.OpenebsNamespace,
+		},
+	}
+	for _, config := range configs {
+		switch config.Name {
+		case "TargetResourceRequests":
+			res, err := parseResource(config.Value)
+			if err != nil {
+				return err
+			}
+			if cvp.Spec.Target.Resources == nil {
+				cvp.Spec.Target.Resources = &corev1.ResourceRequirements{}
+			}
+			cvp.Spec.Target.Resources.Requests = res
+		case "TargetResourceLimits":
+			res, err := parseResource(config.Value)
+			if err != nil {
+				return err
+			}
+			if cvp.Spec.Target.Resources == nil {
+				cvp.Spec.Target.Resources = &corev1.ResourceRequirements{}
+			}
+			cvp.Spec.Target.Resources.Limits = res
+		case "AuxResourceRequests":
+			res, err := parseResource(config.Value)
+			if err != nil {
+				return err
+			}
+			if cvp.Spec.Target.AuxResources == nil {
+				cvp.Spec.Target.AuxResources = &corev1.ResourceRequirements{}
+			}
+			cvp.Spec.Target.AuxResources.Requests = res
+		case "AuxResourceLimits":
+			res, err := parseResource(config.Value)
+			if err != nil {
+				return err
+			}
+			if cvp.Spec.Target.AuxResources == nil {
+				cvp.Spec.Target.AuxResources = &corev1.ResourceRequirements{}
+			}
+			cvp.Spec.Target.AuxResources.Limits = res
+		case "TargetNodeSelector":
+			x := map[string]string{}
+			err = yaml.Unmarshal([]byte(config.Value), &x)
+			if err != nil {
+				panic(err.Error())
+			}
+			cvp.Spec.Target.NodeSelector = x
+		case "TargetTolerations":
+			tMap := map[string]corev1.Toleration{}
+			err = yaml.Unmarshal([]byte(config.Value), &tMap)
+			if err != nil {
+				panic(err.Error())
+			}
+			t := []corev1.Toleration{}
+			for _, y := range tMap {
+				t = append(t, y)
+			}
+			cvp.Spec.Target.Tolerations = t
+		case "Luworkers":
+			cvp.Spec.Target.IOWorkers, err = strconv.ParseInt(config.Value, 10, 64)
+			if err != nil {
+				return err
+			}
+		case "QueueDepth":
+			cvp.Spec.Target.QueueDepth = config.Value
+		case "ZvolWorkers":
+			cvp.Spec.Replica.IOWorkers = config.Value
+		case "FSType":
+			sc.Parameters["fsType"] = config.Value
+		}
+	}
+	sc.Parameters["cstorVolumePolicy"] = cvp.Name
```

Comment:
this should be moved below - post the CVP is successfully created.

Reply:
Done